### PR TITLE
feat: add detail error msg for jdbc insert

### LINKED ISSUE #2506: add detail error msg for jdbc insert
https://github.com/4paradigm/OpenMLDB/blob/053c4e3d885b237711d3719be7bf0c1905c5197e/java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/InsertPreparedStatementImpl.java#L353-L359

For example, `append failed on ${currentDataIdx}(${curType}), current data ${data}`

"build insert row failed(str size init!=actual)"

### PR DESCRIPTION
**Pull request by haseeb-xd**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Modified error message

* **What is the current behavior?** (You can also link to an open issue here)
Issue #2506

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

**Comment by dl239** [comment]
hi @haseeb-xd there are uncorrelated file changes in this pr. it's better to merge the latest main branch.

**Comment by haseeb-xd** [comment]
> hi @haseeb-xd there are uncorrelated file changes in this pr. it's better to merge the latest main branch.

Okay, Will takecare if this happens again.

**Comment by haseeb-xd** [comment]
> > hi @haseeb-xd there are uncorrelated file changes in this pr. it's better to merge the latest main branch.
> 
> Okay, Will takecare if this happens again. other's are previous commits that are already merged.

**Comment by vagetablechicken** [comment]
I've merge main to this branch, only jdbc files changed

**Review by dl239 — APPROVED** [review]

File: java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/InsertPreparedStatementImpl.java [hunk 1]
```diff
@@ -351,11 +351,11 @@ private void buildRow() throws SQLException {
                 }
             }
             if (!ok) {
-                throw new SQLException("append failed");
+                throw new SQLException("append failed on currentDataIdx{curType}, current data"+data);
```
Comment (vagetablechicken) [inline java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/InsertPreparedStatementImpl.java]: plz print the real `currentDataIdx` and `curType`, the msg in issue is just pseudocode

Comment (haseeb-xd) [reply java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/InsertPreparedStatementImpl.java]: curType is defined in "else block",  It has no scope outside that.  I can access currentDataIdx

Comment (vagetablechicken) [reply java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/InsertPreparedStatementImpl.java]: you can move `DataType curType` or just use `currentDatasType.get(currentDataIdx)`

**Comment by vagetablechicken** [comment]
@haseeb-xd  Plz fix the test about exception msg check `SQLRouterSmokeTest.testInsertPreparedStateBatch:510`

**Comment by haseeb-xd** [comment]
> @haseeb-xd Plz fix the test about exception msg check `SQLRouterSmokeTest.testInsertPreparedStateBatch:510`

Can you please explain what msg do you want me to put on line 510?

**Comment by vagetablechicken** [comment]
> > @haseeb-xd Plz fix the test about exception msg check `SQLRouterSmokeTest.testInsertPreparedStateBatch:510`
> 
> Can you please explain what msg do you want me to put on line 510?



> > @haseeb-xd Plz fix the test about exception msg check `SQLRouterSmokeTest.testInsertPreparedStateBatch:510`
> 
> Can you please explain what msg do you want me to put on line 510?

 Assert.assertTrue(e.getMessage().contains("append failed"));

File: java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/InsertPreparedStatementImpl.java [hunk 2]
```diff
@@ -351,11 +351,11 @@ private void buildRow() throws SQLException {
                 }
             }
             if (!ok) {
-                throw new SQLException("append failed");
+                throw new SQLException("append failed on currentDataIdx: "+currentDataIdx+", curType: "+currentDatasType.get(currentDataIdx)+", current data: "+data);
```
Comment (tobegit3hub) [inline java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/InsertPreparedStatementImpl.java]: May add space between plus sign. Take this for example.

```
"append failed on currentDataIdx: " + currentDataIdx + ", curType: " + currentDatasType.get(currentDataIdx) + ", current data: " + data
```

**Review by tobegit3hub — APPROVED** [review]
LGTM

**Pull request merged**